pipeline.hpp

```diff
@@ -137,22 +136,41 @@ class Pipeline {
     void operator=(const Pipeline&) = delete;
     void operator=(Pipeline&&) = delete;
 
+    /// The main processing method that either pick up the waiting producer and send its data
+    /// or wait for notification if all scheduler producers are processed.
     void run_until_stopped();
+    /// Unblock and finish the run_until_stopped method, must be called from different thread than
```

Comment:
Does it really have to be called from different thread?

Reply:
You're right, it can be, but not necessarily.